lib/DBDish/Connection.pm6

```diff
@@ -23,6 +23,23 @@ has Lock $!statements-lock .= new;
 has $.last-sth-id is rw;
 has $.last-rows is rw;
 
+## Introspection methods added to facilitate better testing of statement handle management!
+## While the Oracle specific driver does have this issue I suspect others do as well for
+## the same reasons. Connection methods such as .execute, .do, .commit, .rollback perhaps
+## others have been know to leak handles until Oracle runs out of cursors.
+method inspect-statement-count { return $!statements-lock.protect: { %!statements.keys.elems }}
+method inspect-statement-keys  { return $!statements-lock.protect: { %!statements.keys }}
+method inspect-statement-sql
+{
+  return $!statements-lock.protect: {
+    my @sql-statements;
+    for %!statements.values -> $sth {
+      @sql-statements.push: $sth.statement;
+    }
+    @sql-statements;
+  }
+}
+
```

Comment:
Will they be removed after the issue has been resolved or will they be part of the API? If Raku provides no middle way for testing I'd name them starting with an underscore.

Reply:
Removing them would break the test; besides, if we broke handle management in the beginning who says we wont break it again later. Better to keep on testing. Ever since I embraced writing for test I quickly learned that black boxes are just places for pure evil code to fester; adding introspection just makes sense to me so I do it all the time. To paraphrase Dr. Venkmen (Ghost-busters); no code is too big, no test is too big!

Reply:
There probably is a way of doing this in the tests without them becoming part of the API by applying a test specific role to the object under test that provides the inspection methods - I'll look at that when they are merged.